analyses/molecular-subtyping-HGG/04-HGG-molecular-subtyping-mutation.Rmd

```diff
@@ -0,0 +1,331 @@
+Should they be included in the subset files?_
+
+#### _BRAF_ V600E
+
+```{r}
+braf_snv_df <- filtered_snv_df %>%
+  dplyr::filter(Hugo_Symbol == "BRAF",
+                grepl("V600E", HGVSp_Short))
+
+head(braf_snv_df, n = 10)
+```
+
+### Wrangle SNV Data
+
+Now that we've filtered the data, we'll get it into a form that's more amenable to presentation.
+First, let's include a column that contains all of the coding mutations of interest (excluding _TERT_).
+This will contain duplicated information -- e.g., _H3F3A_ will be present for tumors with _H3F3A_ K28M mutations -- but serve as a "sanity check."
+
+```{r}
+coding_summary_df <- coding_snv_df %>% 
+  dplyr::select(Tumor_Sample_Barcode,
+                Hugo_Symbol) %>%
+  # For each tumor biospecimen, get a comma sepearted
+  dplyr::group_by(Tumor_Sample_Barcode) %>%
+  dplyr::summarize(relevant_coding_mutations = paste(sort(unique(Hugo_Symbol)),
+                                                     collapse = ", "))
+
+head(coding_summary_df, n = 10)
+```
+
+Add tumors that had no mutations in the relevant genes into this `data.frame`.
+
+```{r}
+# This will be all biospecimen IDs that were included in the HGG subset
+# SNV file
+subset_bsid <- unique(snv_df$Tumor_Sample_Barcode)
+# Add the 'missing' biospecimen IDs into the coding summary data.frame but
+# mark relevant_coding_mutations as "None"
+coding_summary_df <- coding_summary_df %>%
+  dplyr::bind_rows(data.frame(
+    Tumor_Sample_Barcode = setdiff(subset_bsid,
+                                   coding_summary_df$Tumor_Sample_Barcode)
+  )) %>%
+  replace(is.na(.), "None")
+```
+
+We'll make a `data.frame` that indicates the Variant Classification of any _TERT_ mutation in a tumor biospecimen.
+
+```{r}
+tert_summary_df <- tert_snv_df %>%
+  dplyr::select(Tumor_Sample_Barcode,
+                Variant_Classification) %>%
+  dplyr::group_by(Tumor_Sample_Barcode) %>%
+  dplyr::summarize(TERT_variant_classification = 
+                     paste(sort(unique(Variant_Classification)),
+                           collapse = ", "))
+head(tert_summary_df)
+```
+
+Add in tumor biospecimens where no _TERT_ mutation is present.
+
+```{r}
+tert_summary_df <- tert_summary_df %>%
+  dplyr::bind_rows(data.frame(
+    Tumor_Sample_Barcode = setdiff(subset_bsid,
+                                   tert_summary_df$Tumor_Sample_Barcode)
+  )) %>%
+  replace(is.na(.), "None")
+```
+
+Now a `data.frame` that contains the specific IDH1 mutations.
+
+```{r}
+idh_summary_df <- idh_snv_df %>%
+  dplyr::select(Tumor_Sample_Barcode, HGVSp_Short) %>%
+  dplyr::rename(IDH1_mutation = HGVSp_Short) %>%
+  dplyr::bind_rows(data.frame(
+    Tumor_Sample_Barcode = setdiff(subset_bsid,
+                                   idh_snv_df$Tumor_Sample_Barcode)
+  )) %>%
+  replace(is.na(.), "No R132 or R172")
+```
+
+And a `data.frame` that contains the specific _BRAF_ mutation.
+
+```{r}
+braf_summary_df <- braf_snv_df %>%
```

Comment:
Because you repeat a similar process for each mutation, it almost feels like this should be a function. However, I am unsure the payoff of making such a function would be that great given the context.

Reply:
I came to the same conclusion, i.e., pay off would not necessarily be high enough to justify potentially making things harder to follow.